### PR TITLE
[risk=low][no ticket] Upgrades to API packages and notes for cases where we can't

### PR DESCRIPTION
**Pull request by jmthibault79**
Tested locally by interacting with workspaces, runtimes, apps, and disks

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/build.gradle [hunk 1]
```diff
@@ -20,23 +20,26 @@ buildscript {
...
         OKHTTP_VERSION = '4.12.0'
         OPENTELEMETRY_SDK_VERSION = '1.43.0'
-        OPENTELEMETRY_JAVA_VERSION = '2.5.0'
+        OPENTELEMETRY_JAVA_VERSION = '2.8.0'
         OPENTELEMETRY_GOOGLE_VERSION = '0.31.0'
         SPRINGFOX_VERSION = '3.0.0'
         SPRING_BOOT_VERSION = '3.3.2'
-        SPRING_FRAMEWORK_VERSION = '6.1.9'
-        SPRING_SECURITY_VERSION = '6.3.0'
-        SWAGGER_3_CODEGEN_VERSION = '3.0.55'
```
Comment (jmthibault79) [inline api/build.gradle:39]: this was only used in one location, so I inlined it.  `SWAGGER_VERSION` refers to other packages.

File: api/build.gradle [hunk 2]
```diff
@@ -420,7 +426,6 @@ dependencies {
     implementation "io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi:$project.ext.OPENTELEMETRY_SDK_VERSION"
     implementation "io.opentelemetry.instrumentation:opentelemetry-instrumentation-api:$project.ext.OPENTELEMETRY_JAVA_VERSION"
     implementation "io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:$project.ext.OPENTELEMETRY_JAVA_VERSION"
-    implementation "io.opentelemetry.instrumentation:opentelemetry-spring-boot-3:$project.ext.OPENTELEMETRY_JAVA_VERSION-alpha"
```
Comment (jmthibault79) [inline api/build.gradle:423]: this appears to be discontinued, and we don't seem to need it

File: api/build.gradle [hunk 3]
```diff
@@ -433,16 +438,17 @@ dependencies {
     }
     implementation "com.google.cloud.opentelemetry:exporter-auto:$project.ext.OPENTELEMETRY_GOOGLE_VERSION-alpha"
 
-    implementation 'io.swagger:swagger-annotations:1.6.14'
```
Comment (jmthibault79) [inline api/build.gradle:436]: replaced by `io.swagger.core.v3:swagger-annotations` - see line 589

File: api/build.gradle [hunk 4]
```diff
@@ -576,22 +586,18 @@ dependencies {
     // Swagger3 codegen deps
     // TODO(calbach): The duplicated deps here are highly suspect. Investigate
     // removing the generated sourceSet, or else fixing the runtime deps.
-    implementation 'io.swagger.core.v3:swagger-annotations:2.2.3'
-    implementation 'io.swagger.core.v3:swagger-models:2.2.3'
-    implementation 'io.gsonfire:gson-fire:1.8.5'
-    implementation 'org.threeten:threetenbp:1.6.2'
```
Comment (jmthibault79) [inline api/build.gradle:582]: `threetenbp` and `springfox-oas` don't appear to be used.

File: api/build.gradle [hunk 5]
```diff
@@ -20,23 +20,25 @@ buildscript {
     // External properties on the default project. Values declared in ext blocks
     // outside of the buildscript block aren't usable here.
     ext {
-        GAE_VERSION = '2.0.29'
-        GOOGLE_TRUTH_VERSION = '1.1.3'
```
Comment (jmthibault79) [inline api/build.gradle:24]: inlined

File: api/build.gradle [hunk 6]
```diff
@@ -576,27 +586,23 @@ dependencies {
...
     providedCompile group: 'jakarta.servlet', name: 'jakarta.servlet-api', version: '6.0.0'
 
-    __tools__Implementation 'commons-cli:commons-cli:1.5.0'
-    __tools__Implementation 'com.opencsv:opencsv:5.7.0'
+    __tools__Implementation 'commons-cli:commons-cli:1.9.0'
+    __tools__Implementation 'com.opencsv:opencsv:5.9'
 
-    testImplementation 'org.mockito:mockito-core:5.3.1'
-    testImplementation "com.google.truth:truth:$project.ext.GOOGLE_TRUTH_VERSION"
-    testImplementation "com.google.truth.extensions:truth-java8-extension:$project.ext.GOOGLE_TRUTH_VERSION"
```
Comment (jmthibault79) [inline api/build.gradle:598]: no longer needed - included in the base truth package

File: api/build.gradle (hunk 4 above)
Comment (jmthibault79) [reply api/build.gradle:582]: or: they are included as subdeps and we don't need to call them out

**Review by yonghaoy — APPROVED** [review]

**Pull request merged**